test/watcher_test.js

```diff
@@ -139,6 +162,65 @@ describe('Watcher', function() {
       expect(builderBuild).to.have.been.called;
     });
 
+    it('on change, rebuild is invoked and cancel is invoked from the build', function() {
+      const pluginsCalled = {};
```

Comment:
On second thought, maybe it would be better to make this an array:

```suggestion
      const events = [];
```

This allows us to do something like this (psuedo code):

```js
const events = [];

class Plugin1 extends Plugin {
  build() {
    events.push('plugin1 build');

    return waitForPlugin1.promise;
  }
}

class Plugin2 extends Plugin {
  build() {
    events.push('plugin2 build');

    return waitForPlugin2.promise;
  }
}

//...snip...
expect(events).to.deepEqual([
  'plugin1 build',
  'plugin2 build',
  'plugin1 build',
  //...etc
]);
```

Changing to this structure lets us assert not only on the exact count of `build` calls for each plugin, but their exact ordering.